File: agora-frontend/src/types/mentorship.ts
export type MentorshipStatus =
  | 'PendingApproval'
  | 'Active'
  | 'Rejected'
  | 'Revoked'
  | 'Completed';

export type MentorshipGoalStatus =
  | 'Pending'
  | 'Submitted'
  | 'Approved'
  | 'NeedsRevision';

export type MentorshipTask = {
  id: number;
  mentorshipGoalId: number;
  title: string;
  description?: string | null;
  isCompleted: boolean;
  completedAt?: string | null;
  createdAt: string;
};

export type MentorshipGoal = {
  id: number;
  mentorshipId: number;
  title: string;
  description: string;
  dueDate?: string | null;
  status: MentorshipGoalStatus;
  statusName: string;
  professorFeedback?: string | null;
  studentSubmissionNote?: string | null;
  createdAt: string;
  completedAt?: string | null;
  reviewedAt?: string | null;
  tasks?: MentorshipTask[];
  totalTasks?: number;
  completedTasks?: number;
  progressPercent?: number;
};

export type MentorshipMessage = {
  id: number;
  mentorshipId: number;
  senderId: string;
  senderName: string;
  senderPhotoUrl?: string | null;
  senderRole: string;
  content: string;
  createdAt: string;
};

export type Mentorship = {
  id: number;
  projectId: number;
  projectTitle: string;
  professorId: string;
  professorName: string;
  professorPhotoUrl?: string | null;
  professorCourse?: string | null;
  studentAuthorId: string;
  studentAuthorName: string;
  status: MentorshipStatus;
  statusName: string;
  initiatedBy: 'Professor' | 'Student';
  initiatedByName: string;
  requestMessage?: string | null;
  requestedAt: string;
  acceptedAt?: string | null;
  endedAt?: string | null;
  totalGoals: number;
  pendingReviewGoals: number;
  completedGoals: number;
  totalTasks?: number;
  completedTasks?: number;
  overallProgressPercent?: number;
  goals: MentorshipGoal[];
};

File: backend/Models/Mentorship.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Nexora.Api.Enums;

namespace Nexora.Api.Models;

public class Mentorship
{
    [Key]
    public int Id { get; set; }

    [Required]
    public int ProjectId { get; set; }

    [ForeignKey("ProjectId")]
    public Project? Project { get; set; }

    [Required]
    public string ProfessorId { get; set; } = string.Empty;

    [ForeignKey("ProfessorId")]
    public ApplicationUser? Professor { get; set; }

    [Required]
    public MentorshipStatus Status { get; set; } = MentorshipStatus.PendingApproval;

    [Required]
    public MentorshipInitiator InitiatedBy { get; set; }

    [MaxLength(1000)]
    public string? RequestMessage { get; set; }

    public DateTime RequestedAt { get; set; } = DateTime.UtcNow;

    public DateTime? AcceptedAt { get; set; }

    public DateTime? EndedAt { get; set; }

    public ICollection<MentorshipGoal> Goals { get; set; } = new List<MentorshipGoal>();
    public ICollection<MentorshipMessage> Messages { get; set; } = new List<MentorshipMessage>();
}
